smart-contract-verifier/src/solidity/solc_cli.rs

```diff
@@ -0,0 +1,493 @@
+//! Module for compiling solidity contracts using cmd args.
+//! ethers_solc compiler uses --standard-json method of input,
+//! because it's very easy and convient, however --standard-json flag
+//! was added only since 0.4.10 version. So, to compile older versions
+//! we need convert functions for CompilerInput and CompilerOutput.
+
+use std::{
+    collections::BTreeMap,
+    path::PathBuf,
+    process::{Command, Stdio},
+};
+
+use ethers_solc::{
+    artifacts::Severity,
+    error::{SolcError, SolcIoError},
+    CompilerInput, CompilerOutput,
+};
+
+mod serde_helpers {
+    use serde::de;
+
+    pub fn deserialize_abi_string<'de, D>(
+        deserializer: D,
+    ) -> Result<Vec<serde_json::Value>, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        let s: String = de::Deserialize::deserialize(deserializer)?;
+        serde_json::from_str(&s).map_err(de::Error::custom)
+    }
+}
+
+mod types {
+    use super::serde_helpers;
+    use std::collections::{BTreeMap, HashMap};
+
+    use ethers_solc::{
+        artifacts::{Contract, Libraries},
+        error::SolcError,
+        CompilerInput, CompilerOutput,
+    };
+    use minicbor::encode::Write;
+    use serde::{Deserialize, Serialize};
+    use tempfile::TempDir;
+
+    #[derive(Debug, PartialEq, Eq)]
+    pub struct InputArgs {
+        pub optimize: bool,
+        pub optimize_runs: Option<usize>,
+        pub libs: BTreeMap<String, String>,
+    }
+
+    fn merge_libs(libraries: Libraries) -> BTreeMap<String, String> {
+        libraries
+            .libs
+            .into_iter()
+            .map(|(_, libs)| libs)
+            .reduce(|accum, item| accum.into_iter().chain(item).collect())
+            .unwrap_or_default()
+    }
```

Comment:
Maybe we may do something like this to make it more clear?

```suggestion
    fn merge_libs(libraries: Libraries) -> BTreeMap<String, String> {
        let mut result = BTreeMap::new();
        libraries
            .libs
            .into_iter()
            .for_each(|(_, libs)| result.extend(libs));
        result
    }
```